### DISCUSSION TITLE
How can I change cursor on top of browser control?

### DISCUSSION THREAD

**Discussion by spacevoyager78** [cefsharp/CefSharp#3369] [Q&A]
Hi!
I would like to display a waiting cursor whenever a page is still loading. However, everything I tried doesn't work on top of browser, but only on top of other controls of my application (winforms).
I am using the `ChromiumWebBrowser.LoadingStateChanged` event and inside it I have tried all of the following:
```
browser.Cursor = args.CanReload ? Cursors.Default : Cursors.WaitCursor; // nothing happens
browser.UseWaitCursor = !args.CanReload; // nothing happens
this.Cursor = args.CanReload ? Cursors.Default : Cursors.WaitCursor; // works only for other form controls
this.UseWaitCursor = !args.CanReload; // works only for other form controls
Application.UseWaitCursor = !args.CanReload; //works only for other form controls
```
I have also discovered the `IBrowserHost.MouseCursorChangeDisabled` prorerty, but that also didn't help.
What am I missing?

**Comment by amaitland** [comment]
> I am using the `ChromiumWebBrowser.LoadingStateChanged` event and inside it I have tried all of the following:

I don't expect that would work as `CEF` runs it's own message loop by default, you can try integrating `CEF` into your applications existing message loop, see https://github.com/cefsharp/CefSharp/wiki/General-Usage#multithreadedmessageloop for some details.

If that doesn't work you might need to grab the `HWND` and call the `Win32` equability function via `P/Invoke`. I haven't actually tried, I'd suggest starting with integrating `CEF` into your applications existing message loop.

> I have also discovered the `IBrowserHost.MouseCursorChangeDisabled` prorerty, but that also didn't help.

Doubt that's relevant.
